contextPlus.js

```diff
@@ -4,78 +4,80 @@ const contextMenuContainers = {
       return;
     }
 
-    const contextualIdentities = await browser.contextualIdentities.query({});
-    const parentId = browser.contextMenus.create({
-      id: "moveContext",
-      title: "Move to Context",
-      contexts: ["tab", "page"]
-    });
+    const defaultCookieStoreId = "firefox-default";
 
+    const contextualIdentities = await browser.contextualIdentities.query({});
     const contextStore = contextualIdentities.reduce((store, context) => {
-      return Object.assign({}, store, {
-        [`contextPlus-${context.name}`]: context.cookieStoreId
-      });
+      return Object.assign(
+        { "contextPlus-default": defaultCookieStoreId },
+        store,
+        { [`contextPlus-${context.name}`]: context.cookieStoreId }
+      );
     }, {});
 
-    // Add a default context manually.
-    // Hope the name of the default cookieStore never changes :)
-    contextStore["contextPlus-default"] = "firefox-default";
-    browser.contextMenus.create({
-      type: "normal",
-      title: "Default",
-      id: "contextPlus-default",
-      parentId
-    });
-
-    const colors = {
-      blue: '00a7e0',
-      turquoise: '01bdad',
-      green: '7dc14c',
-      yellow: 'ffcb00',
-      orange: 'ff9216',
-      red: 'd92215',
-      pink: 'ee5195',
-      purple: '7a2f7a'
-    };
-
-    contextualIdentities.forEach(context => {
-
-      fetch(`icons/usercontext-${context.icon}.svg`).then( response => response.text()).then( svg => {
-        svg = svg.replace('context-fill', '%23'+colors[context.color]);
+    // The context menu is re-created each time a tab is activated
+    // to account for its context.
+    const onActivatedTabHandler = async function({ tabId }) {
+      browser.contextMenus.removeAll();
+      const parentId = browser.contextMenus.create({
+        id: "moveContext",
+        title: "Move to Context",
+        contexts: ["tab", "page"],
+      });
 
+      const activeTab = await browser.tabs.get(tabId);
+      if (activeTab.cookieStoreId !== defaultCookieStoreId) {
         browser.contextMenus.create({
-          type: 'normal',
-          title: context.name,
-          id: `contextPlus-${context.name}`,
+          type: "normal",
+          title: "No Context",
+          id: "contextPlus-default",
           parentId,
-          icons: {
-            16: 'data:image/svg+xml;utf8,'+svg
-          }
         });
-        
-      });
+        browser.contextMenus.create({
+          type: "separator",
+          id: "contextPlus-separator",
+          parentId,
+        });
+      }
 
-    });
+      contextualIdentities
+        .filter(context => context.cookieStoreId !== activeTab.cookieStoreId)
+        .forEach(context => {
+          fetch(`icons/usercontext-${context.icon}.svg`)
+            .then(response => response.text())
+            .then(svg => {
+              svg = svg.replace(
+                "context-fill",
+                context.colorCode.replace("#", "%23")
```

Comment:
This `colorCode` attribute isn't available in Stable, so we can't yet ditch the colors dictionary.